src/common/OpQueue.h

```diff
@@ -37,10 +37,15 @@ class OpQueue {
   public:
     // How many Ops are in the queue
     virtual unsigned length() const = 0;
-    // Ops will be removed f evaluates to true, f may have sideeffects
+    // Ops will be removed f evaluates to true, f may have side effects.
+    // Must visit operations from back to front, due to its use with
+    // OSD::SharedOpQueue::Pred .
     virtual void remove_by_filter(
```

Comment:
OK, I'll clean up the OpQueue interface to remove that pure virtual method and clean up the subclasses. I use that function in mClockPrioirtyQueue, though, so I'll keep it there. It's how the two integrations bridge certain functionality.